### ISSUE TITLE
[BUG] Plot failing randomly 

### ISSUE THREAD

**Issue by MasterAssailant** [Chia-Network/chia-blockchain#2951]
**Describe the bug**
I am using GUI to plot. After restarting the GUI it will be able to create a few plots successfully. Then it will randomly fail (i.e. I have seen it fail from 5X% up to 9x%) without showing any error in the log. It just got stuck with no further update in the log and cpu and disk activities back to normal. Have to restart and clear everything to plot until it fails again. It happened when I try to plot both in sequential or in parallel. The plotting SSD doesn't seem to be overheating. The plotting setting I am using is 6750MB of memory with 4 threads for each queue. I have tried using the default value but the problem seems to persist. 

Here's a sample of the last entries from the log from the most recent crash:

Stuck at 88%
```
	Bucket 85 uniform sort. Ram: 6.525GiB, u_sort min: 1.500GiB, qs min: 0.479GiB.
	Bucket 86 QS. Ram: 6.525GiB, u_sort min: 0.750GiB, qs min: 0.319GiB. force_qs: 1
	Second computation pass time: 398.575 seconds. CPU (99.940%) Wed Apr 28 01:50:33 2021
	Wrote 3715555769 entries
Total compress table time: 1168.620 seconds. CPU (99.150%) Wed Apr 28 01:50:33 2021
Compressing tables 6 and 7
	Bucket 0 uniform sort. Ram: 6.525GiB, u_sort min: 0.500GiB, qs min: 0.250GiB.
	Bucket 1 uniform sort. Ram: 6.525GiB, u_sort min: 0.500GiB, qs min: 0.250GiB.
	...
	Bucket 58 uniform sort. Ram: 6.525GiB, u_sort min: 0.500GiB, qs min: 0.250GiB.
```

Stuck at 87%
```
	Bucket 85 uniform sort. Ram: 6.525GiB, u_sort min: 1.500GiB, qs min: 0.479GiB.
	Bucket 86 QS. Ram: 6.525GiB, u_sort min: 0.750GiB, qs min: 0.325GiB. force_qs: 1
	Second computation pass time: 428.006 seconds. CPU (95.290%) Tue Apr 27 23:07:18 2021
	Wrote 3716059261 entries
Total compress table time: 1404.048 seconds. CPU (83.810%) Tue Apr 27 23:07:18 2021
Compressing tables 6 and 7
	Bucket 0 uniform sort. Ram: 6.525GiB, u_sort min: 0.500GiB, qs min: 0.250GiB.
	...
	Bucket 28 uniform sort. Ram: 6.525GiB, u_sort min: 0.500GiB, qs min: 0.250GiB.
```

In recent fails I noticed there is an application error entry shown in the windows event viewer. But the latest occurrence there is no such error entry. 

```
Faulting application name: chia.exe, version: 0.0.0.0, time stamp: 0x5ffec122
Faulting module name: chiapos.cp37-win_amd64.pyd, version: 0.0.0.0, time stamp: 0x606797c8
Exception code: 0xc0000005
Fault offset: 0x0000000000001800
Faulting process id: 0x6a10
Faulting application start time: 0x01d73a5de424abba
Faulting application path: C:\Users\XXX\AppData\Local\chia-blockchain\app-1.1.2\resources\app.asar.unpacked\daemon\chia.exe
Faulting module path: C:\Users\XXX\AppData\Local\chia-blockchain\app-1.1.2\resources\app.asar.unpacked\daemon\chiapos.cp37-win_amd64.pyd
Report Id: a0eb4f7f-6e84-4b93-a317-f65600e20e40
Faulting package full name: 
Faulting package-relative application ID: 
```

This bug is present from version 1.0.X till the latest version 1.1.2.

**To Reproduce**

1. Perform plotting via GUI.
2. After finishing maybe 1 to 2 plots then fail.

**Expected behavior**
Plotting stops, no more log update with no error shown in the log. CPU and Disk activities back to normal.

**Screenshots**
(Just normal plotting GUI screen with the status stuck at "Plotting" and showing the percentage.)

**Desktop (please complete the following information):**

- OS: Windows 10 
- OS Version/Flavor: 20H2 19042.928
- CPU: i9 10900K
- RAM 64GB
- Plotting Temp Disk: Samsung 980 EVO 1TB
- Destination Disk: WD Red 4TB

**Additional context**
Just saw another issue about mem leak. Wondering if this can be a possible cause for this issue? 
I have also tried closing other programs on my computer while plotting but this problem still exists. 
I have other lower spec computer plotting as well but the problem only exists on this computer. Can this be a problem with the plotting disk 980 evo?
Also I noticed the number of process shown on task manager is quite a lot not sure if this is normal...
![Usage](https://user-images.githubusercontent.com/29315038/116330022-35708e00-a7ff-11eb-8e84-7a2dfdca4461.png)

Thank you so much in advance!

**Comment by MasterAssailant** [comment]
I have just tried making another two plots with CLI. It fails again, and this time I managed to capture the system behaviour when it is failing. Hope this can give a better insight into why the plotting will fail:

![Fail Cropped](https://user-images.githubusercontent.com/29315038/116368908-62db2d00-a83b-11eb-97f9-4e19d378173e.png)

The PowerShell on the left is the first one and started a few hours before the second queue on the right. Both failed at a similar phase (phase 3/4) and when compressing table 6 and 7, around bucket 5x to 8x. Previous fails may have halted at similar stages. My guess is that would this be a problem of not having sufficient SSD/system cache or sth related?

**Comment by razamataz** [comment]
+

**Comment by kenelf** [comment]
I have exactly same problem, all failed at 3rd phase, compressing tables

**Comment by kenelf** [comment]
I got 2 SSD as temp. 1 is Samsung 980 pro 1 TB, 1 is Samsung 970 evo plus. so I don't think it is caused by the SSD, because I used the 970 Evo plus to plot on my low-speed PC as an external SSD, never have a problem even it became very hot. 

My PC set up is Ryzen R5 5600X  32GB AKATA mem, I think the cash or mem is enough for plotting, just don't know where this problem comes from. Newly installed system to plot, never success so far, all fail with same problem

**Comment by razamataz** [comment]
I'm not sure if it helped, but for over a day without a glitch. On the chia process, I prioritized "Below Average".

**Comment by kenelf** [comment]
I found the reason for my problem. I ran a mem test, lots of errors showed up, then I googled and suspected it may be caused by overclock, so I set the mem clock back to default 2666 instead of 3600 which auto set by the motherboard. After that, the problems are gone, now I can plot smoothly. Recommend you check your mem.

**Comment by razamataz** [comment]
i test my Ram first and it OK.

**Comment by kazan417** [comment]
Same behaviour on my system/ stuck at compressing table 6 and 7

**Comment by krasher9** [comment]
I am facing this issue with one of my computers that plots..  
I get no Error just stuck at 88% and when i open to see the info its just stuck at bucket 55.

This happened multiple times , kinda annoying especially when its almost done....

Hopefully i can find out whats the issue, just for your references i use

AMD Ryzen 9 5900X   (24/12) 
Asrock 570x Razor Edition 
3 m.2  -  ( 250gb for windows)  .  2x 980 evo 2TB  - raided to 4TB.
6 HDDS  - 14TBS IRONWOLF
64GB RAM GSKILL 3600MHZ

**Comment by kenelf** [comment]
> I am facing this issue with one of my computers that plots..
> I get no Error just stuck at 88% and when i open to see the info its just stuck at bucket 55.
> 
> This happened multiple times , kinda annoying especially when its almost done....
> 
> Hopefully i can find out whats the issue, just for your references i use
> 
> AMD Ryzen 9 5900X (24/12)
> Asrock 570x Razor Edition
> 3 m.2 - ( 250gb for windows) . 2x 980 evo 2TB - raided to 4TB.
> 6 HDDS - 14TBS IRONWOLF
> 64GB RAM GSKILL 3600MHZ

Try to lower the memory to 3200MHZ or even 2666MHZ, I solved my problem by doing so, hope it helps you too

**Comment by kazan417** [comment]
After addind RAM, now i have 6gb, and increasing swap file to 16gb, i finally finished my first plot.

**Comment by MasterAssailant** [comment]
> > I am facing this issue with one of my computers that plots..
> > I get no Error just stuck at 88% and when i open to see the info its just stuck at bucket 55.
> > This happened multiple times , kinda annoying especially when its almost done....
> > Hopefully i can find out whats the issue, just for your references i use
> > AMD Ryzen 9 5900X (24/12)
> > Asrock 570x Razor Edition
> > 3 m.2 - ( 250gb for windows) . 2x 980 evo 2TB - raided to 4TB.
> > 6 HDDS - 14TBS IRONWOLF
> > 64GB RAM GSKILL 3600MHZ
> 
> Try to lower the memory to 3200MHZ or even 2666MHZ, I solved my problem by doing so, hope it helps you too

To those who are still facing this issue, I have also tried this method, and now the plotting is working flawlessly (a few rounds of plotting without failure). Shoutout to @kenelf for his brilliant solution (tho I am still not sure why a high mem clock rate will cause issues in the first place LOL)!

**Comment by kenelf** [comment]
> > > > Я столкнулся с этой проблемой с одним из моих компьютеров, который строит график.
> > > > Я не получаю ошибки, просто застрял на 88%, и когда я открываю, чтобы увидеть информацию, она просто застревает в корзине 55.
> > > > Это происходило несколько раз, что немного раздражает, особенно когда это почти готово ....
> > > > Надеюсь, я смогу выяснить, в чем проблема, просто для справки я использую
> > > > AMD Ryzen 9 5900X (24/12)
> > > > Asrock 570x Razor Edition
> > > > 3 m.2 - ( 250 ГБ для Windows). 2x 980 evo 2 ТБ - с рейдом до 4 ТБ.
> > > > 6 HDDS - 14TBS IRONWOLF
> > > > 64GB RAM GSKILL 3600MHZ
> > > 
> > > 
> > > Попробуйте уменьшить память до 3200 МГц или даже 2666 МГц, я решил этим свою проблему, надеюсь, это вам тоже поможет
> > 
> > 
> > Тем, кто все еще сталкивается с этой проблемой, я также попробовал этот метод, и теперь построение графика работает безупречно (несколько раундов построения без сбоев). Приветствуем @kenelf за его блестящее решение (хотя я до сих пор не уверен, почему высокая тактовая частота памяти вообще вызовет проблемы, LOL)!
> 
> I have a memory frequency of 2400, should I lower the value too?

If you have a problem with plotting, you can try it.

**Comment by Chinasit316** [comment]
Plz tell me, how to underclock ram. thx bro

**Comment by kenelf** [comment]
you need to get into the motherboard BIOS setting. you can google how to change the RAM clock

**Comment by Chinasit316** [comment]
now i set to DDR4-1600MHz but still error

**Comment by madmmmax** [comment]
2133,2666,3200,3600,freeziiiiiiiiiiiiiiiiiiing
